rotkehlchen/chain/gnosis/modules/sdai/decoder.py

```diff
@@ -0,0 +1,114 @@
+from typing import TYPE_CHECKING, Any
+
+from rotkehlchen.chain.ethereum.utils import token_normalized_value_decimals
+from rotkehlchen.chain.evm.decoding.constants import (
+    CPT_SDAI,
+    SDAI_CPT_DETAILS,
+    SDAI_DEPOSIT,
+    SDAI_REDEEM,
+)
+from rotkehlchen.chain.evm.decoding.interfaces import DecoderInterface
+from rotkehlchen.chain.evm.decoding.structures import (
+    DEFAULT_DECODING_OUTPUT,
+    DecoderContext,
+    DecodingOutput,
+)
+from rotkehlchen.chain.evm.decoding.types import CounterpartyDetails
+from rotkehlchen.chain.gnosis.modules.sdai.constants import GNOSIS_SDAI_ADDRESS
+from rotkehlchen.history.events.structures.types import HistoryEventSubType, HistoryEventType
+from rotkehlchen.types import ChecksumEvmAddress
+from rotkehlchen.utils.misc import hex_or_bytes_to_int
+
+if TYPE_CHECKING:
+    from rotkehlchen.chain.evm.decoding.base import BaseDecoderTools
+    from rotkehlchen.chain.gnosis.node_inquirer import GnosisInquirer
+    from rotkehlchen.user_messages import MessagesAggregator
+
+
+class SdaiDecoder(DecoderInterface):
+    def __init__(
+            self,
+            gnosis_inquirer: 'GnosisInquirer',
+            base_tools: 'BaseDecoderTools',
+            msg_aggregator: 'MessagesAggregator',
+    ) -> None:
+        super().__init__(
+            evm_inquirer=gnosis_inquirer,
+            base_tools=base_tools,
+            msg_aggregator=msg_aggregator,
+        )
+
+    def _decode_sdai_deposit_events(self, context: DecoderContext) -> DecodingOutput:
+        # 18 is the number of decimals of sdai/xdai/wxdai
```

Comment:
both here and in the other parts where you do this import `DEFAULT_TOKEN_DECIMALS` and use that.
And add this comment where you import it. Also probably change it to evm and not ethereum specific as location of constant.

so something like

```python
from rotkehlchen.chain.evm.constants import DEFAULT_TOKEN_DECIMALS  # all tokens in this decoder use default(18)  # noqa: E501
```
Also `DEFAULT_TOKEN_DECIMALS` is not Final. Make it.

This way you can remove the manual 18 literals and the repeating comment